src/main/java/com/textmessenger/service/MessageServiceJpa.java

```diff
@@ -0,0 +1,45 @@
+package com.textmessenger.service;
+
+import com.textmessenger.model.Message;
+import com.textmessenger.repository.MessageRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+public class MessageServiceJpa implements MessageService {
+
+  final MessageRepository messageRepository;
```

Comment:
Where is your autowired annotation

Reply:
Should be put above constructor

Reply:
+ private

Reply:
spring 5.0 autowired is deprecated in next version